ext/storage_sources/s3_store/test/test_s3_connection.cpp

```diff
@@ -325,6 +327,54 @@ TestObjectExists(const Aws::S3Crt::ClientConfiguration &config)
     return (TEST_SUCCESS);
 }
 
+/*
+ * TestBadBucket --
+ *     Unit test to check if connection to a non-existing bucket fails gracefully.
+ */
+int
+TestBadBucket(const Aws::S3Crt::ClientConfiguration &config)
+{
+    int ret = TEST_FAILURE;
+
+    /* The connection object instantitation should not succeed. */
+    try {
+        S3Connection conn(config, "BadBucket", TestDefaults::objPrefix);
+        (void)conn;
```

Comment:
Is this necessary?

Reply:
Ya, unfortunately the compiler complaints that ```conn``` object was not used.